test/Microsoft.AspNetCore.Mvc.Core.Test/ModelBinding/ModelBinderFactoryTest.cs

```diff
@@ -67,9 +67,46 @@ public void CreateBinder_CreatesNoOpBinder_WhenPropertyDoesntHaveABinder()
         }
 
         [Fact]
+        public void CreateBinder_CreatesNoOpBinder_WhenPropertyBindingIsNotAllowed()
+        {
+            // Arrange
+            var metadataProvider = new TestModelMetadataProvider();
+            metadataProvider
+                .ForProperty<Widget>(nameof(Widget.Id))
+                .BindingDetails(m => m.IsBindingAllowed = false);
+
+            var modelBinder = Mock.Of<IModelBinder>();
```

Comment:
Can avoid the mocking e.g. `new ByteArrayModelBinder()` since it just doesn't matter.